app/javascripts/main/store.ts

```diff
@@ -10,15 +17,115 @@ export enum StoreKeys {
   BackupsDisabled = 'backupsDisabled',
   MinimizeToTray = 'minimizeToTray',
   ZoomFactor = 'zoomFactor',
-  /**
-   * @type {Set<String> | null} a set of language codes.
-   */
   SelectedSpellCheckerLanguageCodes = 'selectedSpellCheckerLanguageCodes'
-};
+}
+
+interface StoreData {
+  [StoreKeys.ExtServerHost]: string | null;
+  [StoreKeys.UseSystemMenuBar]: boolean;
+  [StoreKeys.MenuBarVisible]: boolean;
+  [StoreKeys.BackupsLocation]: string;
+  [StoreKeys.BackupsDisabled]: boolean;
+  [StoreKeys.MinimizeToTray]: boolean;
+  [StoreKeys.ZoomFactor]: number;
+  [StoreKeys.SelectedSpellCheckerLanguageCodes]: Set<Language> | null;
+}
+
+function createSanitizedStoreData(data: any = {}): StoreData {
+  return {
+    [StoreKeys.MenuBarVisible]:
+      typeof data[StoreKeys.MenuBarVisible] === 'boolean'
```

Comment:
If not boolean, what other kind of value would we be expecting here?

Reply:
Oh I suppose the value could be `false`, and so you're protecting against that. In any case this would make a good candidate for a helper func, `isBool` or something. It's not ideal typing out 'boolean' in free-form string format every time.

Reply:
As a result of writing tests and thinking about what could break the store (and thus the app,) the most obvious candidate was having corrupt/semantically incorrect data read blindly from the user's hard drive. It has prompted these changes in the store, to treat data stored on disk as unreliable as much as is reasonably possible. So to answer your question, while the code only stores booleans (as enforced by the compiler,) I don't know what else we could expect there, which is the point :) Maybe I'm being too cautious here but as long as we don't have a huge list of things to store in this preferences file, this makes the store more robust. I mean, who knows what antivirus programs are up to these days… I have seen some of them write things to disk _non-stop_ on some computers.